src/ntcore_c.cpp

```diff
@@ -436,3 +436,174 @@ void NT_DisposeRpcCallInfo(NT_RpcCallInfo *call_info) {
+	return retVal;
+}
+
+struct NT_Value *NT_SetNTValueDouble(const char *name, size_t name_len, double v_double, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_InitValue(value);
+	value->data.v_double = v_double;
+	value->type = NT_Type::NT_DOUBLE;
+
+	*set_value_ret = NT_SetEntryValue(name, name_len, value);
+	return value;
+}
+
+struct NT_Value *NT_SetNTValueBoolean(const char *name, size_t name_len, int v_boolean, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_InitValue(value);
+	value->data.v_boolean = v_boolean;
+	value->type = NT_Type::NT_BOOLEAN;
+
+	*set_value_ret = NT_SetEntryValue(name, name_len, value);
+	return value;
+}
+
+struct NT_Value *NT_SetNTValueString(const char *name, size_t name_len, struct NT_String v_string, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_InitValue(value);
+	value->data.v_string = v_string;
+	value->type = NT_Type::NT_STRING;
+
+	*set_value_ret = NT_SetEntryValue(name, name_len, value);
+	return value;
+}
+
+struct NT_Value *NT_SetNTValueRaw(const char *name, size_t name_len, struct NT_String v_raw, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_InitValue(value);
+	value->data.v_raw = v_raw;
+	value->type = NT_Type::NT_RAW;
+
+	*set_value_ret = NT_SetEntryValue(name, name_len, value);
+	return value;
+}
+
+
+
+struct NT_Value *NT_SetNTValueBooleanArr(const char *name, size_t name_len, int *arr, size_t size, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	value->type = NT_Type::NT_DOUBLE_ARRAY;
+	value->data.arr_boolean.arr = arr;
+	value->data.arr_boolean.size = size;
+	return value;
+}
+
+struct NT_Value *NT_CreateNTValueDoubleArr(const char *name, size_t name_len, double *arr, size_t size, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	value->type = NT_Type::NT_DOUBLE_ARRAY;
+	value->data.arr_double.arr = arr;
+	value->data.arr_double.size = size;
+	return value;
+}
+
+struct NT_Value *NT_SetNTValueNTStringArr(const char *name, size_t name_len, struct NT_String *arr, size_t size, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	value->type = NT_Type::NT_DOUBLE_ARRAY;
+	value->data.arr_string.arr = arr;
+	value->data.arr_string.size = size;
+	return value;
+}
+
+void NT_GetDataFromNTValue(struct NT_Value *value,
+	void *data, void **data_arr, size_t *size, enum NT_Type *type,
+	unsigned long long *last_change)
+{
+	*type = value->type;
+	*last_change = value->last_change;
+	switch (value->type)
+	{
+	case NT_Type::NT_BOOLEAN:
+		data = &value->data.v_boolean;
```

Comment:
Almost. You want to be doing
```c
*((int*)data) = value->data.v_boolean;
```
and cast to the appropriate types for the other ones. Also, we should be setting size to the size of what we are writing, so in this case
```c
*size = sizeof(value->data.v_boolean);
```